pkg/kubelet/config/common.go

```diff
@@ -55,17 +57,33 @@ func generatePodName(name string, nodeName types.NodeName) string {
 
 func applyDefaults(pod *api.Pod, source string, isFile bool, nodeName types.NodeName) error {
 	if len(pod.UID) == 0 {
-		hasher := md5.New()
-		hash.DeepHashObject(hasher, pod)
-		// DeepHashObject resets the hash, so we should write the pod source
-		// information AFTER it.
+		var byte_source []byte
+		var errReadSource error
+
 		if isFile {
-			fmt.Fprintf(hasher, "host:%s", nodeName)
-			fmt.Fprintf(hasher, "file:%s", source)
+			byte_source, errReadSource = ioutil.ReadFile(source)
```

Comment:
```suggestion
			byte_source, errReadSource = os.ReadFile(source)
```

Reply:
UID is set by 'pod.UID = types.UID(hex.EncodeToString(hasher.Sum(nil)[0:]))'.
It will change due to a change in the pod structure, which triggers a static pod restart